### PR TITLE
added ManagedExecutor test helper

### PR DESCRIPTION
**Pull request by sanderr**
# Description

Adds a test helper class for scheduler testing. `ManagedExecutor` expects to be driven by the test case itself. It hangs at the start of deploy on a future for which the result should be set by the test case. You can find usage examples in #8131.

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: tests/agent_server/deploy/test_scheduler_agent.py [hunk 1]
```diff
@@ -71,14 +72,53 @@ async def open_version(self, version: int) -> None:
...
+        return self._deploys
+
+    async def stop(self) -> None:
+        # resolve hanging futures to prevent test hanging during teardown
+        for deploy in self._deploys.values():
+            deploy.set_result(const.ResourceState.undefined)
+
+    async def execute(self, gid: uuid.UUID, resource_details: ResourceDetails, reason: str) -> const.ResourceState:
+        assert resource_details.rid not in self._deploys
+        self._deploys[resource_details.rid] = asyncio.Future()
```
Comment (wouterdb) [inline tests/agent_server/deploy/test_scheduler_agent.py:105]: didn't know you could do this, I expect to have to use the create_future of the ioloop.

**Review by wouterdb — APPROVED** [review]
pretty

File: tests/agent_server/deploy/test_scheduler_agent.py (hunk 1 above)
Comment (sanderr) [reply tests/agent_server/deploy/test_scheduler_agent.py:105]: You can do this, but I checked the docs just now and it turns out that they recommend your approach: "the recommended way to create a Future object is to call [loop.create_future()](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.create_future)".

Comment (sanderr) [reply tests/agent_server/deploy/test_scheduler_agent.py:105]:
```suggestion
        self._deploys[resource_details.rid] = asyncio.get_running_loop().create_future()
```

**Comment by jptrindade** [comment]
Could you add a simple test to show how we should use it?

**Review by jptrindade — APPROVED** [review]

**Pull request merged**